### PR TITLE
Remove --transport flag

### PR DESCRIPTION
**Pull request by rhatdan**
This is no simpler then putting the transport in the image page,
we should default to the registry specified in containers/image
and not override it.  People are confused by this option, and I
see no value.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
https://github.com/projectatomic/buildah/issues/108

File: contrib/completions/bash/buildah [hunk 1]
```diff
@@ -279,6 +279,9 @@ return 1
          -*)
              COMPREPLY=($(compgen -W "$boolean_options $options_with_args" -- "$cur"))
              ;;
+         *)
```
Comment (TomSweeneyRedHat) [inline contrib/completions/bash/buildah]: Didn't these changes for the bash completions go in with a different PR?

**Comment by rhatdan** [comment]
I think we might have some other fixes to do to make this work.

@nalind Can I do a 

buildah from registry.access.redhat.com/fedora/rawhide

**Comment by mtrmac** [comment]
This patch requires everyone to say **`docker://`**`registry.access.redhat.com/fedora/rawhide`.  I don’t mind :) but I guess interacting with docker/distribution registries by default is considered desirable?

#229 was not really about the flag being useless, rather about being unintuitively named.

**Comment by rhatdan** [comment]
No I think we should allow the configuration to state where the content will go.  We can default to docker://  But adding

--registry registry.access.redhat.com fedora/rawhide 
Is not a great situation.  

I would prefer we follow the skopeo model, with a default transport being stored in a config file. not needing to have it specified on the command line.

**Comment by mtrmac** [comment]
`--registry` is _not_ being used like `--registry registry.access.redhat.com fedora/rawhide`; that’s one of the ways it is confusing. It is used to specify the containers/image _transport_ (docker/oci-archive/oci-layout/atomic/containers-storage… , along with a possible prefix of the transport-specific location (which may be the `//` syntax, or perhaps a host name, or perhaps even more).

Skopeo actually always requires the transport to be specified; maybe `atomic` does not.

(Also see `kpod`; IIRC some of the commands try to use the input as transport:description, or `docker://`description, depending on what works. I’m not at all sure this is semantically sound / unambiguous, but if various parts of the CLI have to have ambiguous / guessing behavior, it would be nice for all of them to behave consistently.)

**Comment by rhatdan** [comment]
Ok, I think we should default to the docker:// transport all over the place, and allow users to override by specifying the transport in the image name.  Not on some random misunderstood option.

**Comment by TomasTomecek** [comment]
@rhatdan that would make it clear and predictable

**Comment by rhatdan** [comment]
@nalind PTAL

File: docs/buildah-from.md [hunk 2]
```diff
@@ -47,13 +41,13 @@ If an image needs to be pulled from the registry, suppress progress output.
...
 
-buildah from imagename --pull --transport "docker://myregistry.example.com/"
+buildah from imagename --pull
 
 buildah from docker://myregistry.example.com/imagename --pull
 
 buildah from imagename --signature-policy /etc/containers/policy.json
 
-buildah from imagename --pull-always --transport "docker://myregistry.example.com/" --name "mycontainer"
+buildah from docker://myregistry.example.com/imagename --pull-always --name "mycontainer"
```
Comment (TomasTomecek) [inline docs/buildah-from.md]: It would be awesome to point to skopeo's manpage wrt different transports.

Comment (rhatdan) [inline docs/buildah-from.md]: We have added some of that information to buildah man pages, but we could probably add more.

Comment (rhatdan) [inline docs/buildah-from.md]: Checkout buildah-push man page.

Comment (TomasTomecek) [inline docs/buildah-from.md]: That man page rocks! I think you should mention the push man page from man pages which operate on images.

Comment (rhatdan) [inline docs/buildah-from.md]: I could also cut and paste it into the other man pages.

**Comment by rhatdan** [comment]
@TomasTomecek Adding transport information to buildah from man page.  PTAL

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @nalind PTAL

**Comment by rhatdan** [comment]
@nalind Can I merge this?

**Comment by rhatdan** [comment]
@nalind PTAL

File: docs/buildah-from.md (hunk 2 above)
Comment (mtrmac) [reply docs/buildah-from.md]: > I could also cut and paste it into the other man pages.

Eventually there probably should be a centralized man page, something like `container-transports(5)`, which everything could refer to; as we add new transports (or worse, change the semantics of existing transports, as is now happening with the `oci` image tag→reference name), there would be only one page to update instead of dozens; and if that man page lived in containers/image, it could be updated by the PR which changes the implementation.  Only… no such page exists yet I’m afraid.

**Comment by nalind** [comment]
Alright, let's merge this.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 5b596a9
